frontend/src/component/changeRequest/ChangeRequest/Changes/Change/strategy-change-diff-calculation.ts

```diff
@@ -53,80 +58,48 @@ export const getChangesThatWouldBeOverwritten = (
     change: IChangeRequestUpdateStrategy,
 ): ChangesThatWouldBeOverwritten | null => {
     const { snapshot } = change.payload;
+    if (!snapshot || !currentStrategyConfig) return null;
 
-    if (snapshot && currentStrategyConfig) {
-        // compare each property in the snapshot. The property order
-        // might differ, so using JSON.stringify to compare them
-        // doesn't work.
-        const changes: ChangesThatWouldBeOverwritten = Object.entries(
-            omit(currentStrategyConfig, 'strategyName'),
-        )
-            .map(([key, currentValue]: [string, unknown]) => {
-                const snapshotValue = snapshot[key as keyof IFeatureStrategy];
-                const changeValue =
-                    change.payload[key as keyof ChangeRequestEditStrategy];
+    const changes: ChangesThatWouldBeOverwritten = Object.entries(
+        omit(currentStrategyConfig, 'strategyName'),
+    )
+        .map(([key, currentValue]: [string, unknown]) => {
```

Comment:
can we map the function so that it's possible to understand the high level steps of the algorithm without reading the contents of the mapping function?

Reply:
For sure. I was considering that. Will look into it.

Reply:
The typing there gets tricky, but we can use unknown types to get around it. Addressed in 42b70ecebf5a01207007ccf71f9891592d0c4d61